test/ServiceRegistry.js

```diff
@@ -50,1495 +54,1854 @@ describe("ServiceRegistry", function () {
-            await expect(
-                serviceRegistry.connect(account).changeOwner(account.address)
-            ).to.be.revertedWith("OwnerOnly");
-
-            // Trying to change owner for the zero address
-            await expect(
-                serviceRegistry.connect(owner).changeOwner(AddressZero)
-            ).to.be.revertedWith("ZeroAddress");
-
-            // Changing the owner
-            await serviceRegistry.connect(owner).changeOwner(account.address);
-
-            // Trying to change owner from the previous owner address
-            await expect(
-                serviceRegistry.connect(owner).changeOwner(owner.address)
-            ).to.be.revertedWith("OwnerOnly");
-        });
-
-        it("Changing drainer", async function () {
-            const owner = signers[0];
-            const account = signers[1];
-
-            // Trying to change owner from a non-owner account address
-            await expect(
-                serviceRegistry.connect(account).changeDrainer(account.address)
-            ).to.be.revertedWith("OwnerOnly");
-
-            // Trying to change owner for the zero address
-            await expect(
-                serviceRegistry.connect(owner).changeDrainer(AddressZero)
-            ).to.be.revertedWith("ZeroAddress");
-
-            // Changing the owner
-            await serviceRegistry.connect(owner).changeDrainer(account.address);
-
-            // Verifying the drainer address
-            expect(await serviceRegistry.drainer()).to.equal(account.address);
-        });
-
-        it("Should fail when checking for the service id existence", async function () {
-            const tokenId = 0;
-            expect(await serviceRegistry.exists(tokenId)).to.equal(false);
-        });
-
-        it("Should fail when trying to change the serviceManager from a different address", async function () {
-            await expect(
-                serviceRegistry.connect(signers[3]).changeManager(signers[3].address)
-            ).to.be.revertedWith("OwnerOnly");
-        });
-
-        it("Setting the base URI", async function () {
-            await expect(
-                serviceRegistry.connect(signers[1]).setBaseURI("https://localhost2/service/")
-            ).to.be.revertedWith("OwnerOnly");
-
-            await expect(
-                serviceRegistry.setBaseURI("")
-            ).to.be.revertedWith("ZeroValue");
-
-            await serviceRegistry.setBaseURI("https://localhost2/service/");
-            expect(await serviceRegistry.baseURI()).to.equal("https://localhost2/service/");
+        serviceRegistryImplementations.forEach(async function (serviceRegistryImplementation) {
+            it("Changing owner", async function () {
+                // Choose the service registry implementation: L1 or L2
+                if (serviceRegistryImplementation == "l2") {
+                    serviceRegistry = serviceRegistryL2;
+                }
```

Comment:
The service registry contract instance is adjusted depending on the L1 or L2 run